open-craft/edx-platform
# Add custom HTML templates for components in Studio

[SE-4781] This creates a custom list of HTML elements to add to a Unit, in a Subsection, in a Section of a course authored in Studio. This affects any author of courses using Studio.

# JIRA tickets:
SE-4781

# Screenshots:
When clicking HTML (below)
![image](https://user-images.githubusercontent.com/2333868/133158932-9a77a0e3-df3f-4632-a45b-704b430720da.png)
The list that appears (default below)
![image](https://user-images.githubusercontent.com/2333868/133159012-3e6ab788-cc58-4527-849d-b87938c5558c.png)

Should show all the units designed by ESME and available at this [link](https://studio.esme-staging.opencraft.hosting/course/course-v1:esme+CT+2021?show=block-v1%3Aesme%2BCT%2B2021%2Btype%40chapter%2Bblock%400d4aefe059784786bd1f8eb850a633b2)

# Merge deadline: 2021-09-17

# Testing instructions:

1. Make sure to have devstack up and running,
2. In the edx-platform repository pull the branch
3. Run make dev.restart-devserver.studio
4. Visit http://localhost:18010
5. Edit the demo course and add another unit
6. In the new unit add an HTML block
7. Check the items in the list. They need to include the ones added by this commit.
8. Select "overview" from the list
9. Publish the changed course
10. Run make lms-up 
11. Run make lms-static
12. Run frontend-app-learning-up
13. Visit http://localhost:18000, access the demo course as audit student, go to the changed section
14. Check that the overview contains the styles from .div-overview from the scss file changed by this PR

# Author notes and concerns:
This PR is being compared to the current branch in the staging environment.

# Reviewers
@jvdm
